charmhelpers/contrib/openstack/context.py

```diff
@@ -871,23 +872,31 @@ def __call__(self):
         if not self.external_ports or not https():
             return {}
 
-        self.configure_ca()
+        use_keystone_ca = True
+        for rid in relation_ids('certificates'):
+            if related_units(rid):
+                use_keystone_ca = False
```

Comment:
Shortcut here might be

use_keystone_ca = not relation_ids('certificates')

I think just checking for relation presence is OK